### ISSUE TITLE
Editor stops on debugger in typeScriptServices.js:2132

### ISSUE THREAD

**Issue by keaukraine** [microsoft/monaco-editor#1937]
When browser's dev console is open, editor stops on `debugger` keyword in typeScriptServices.js:2132 (node_modules\monaco-editor\esm\vs\language\typescript\lib\typescriptServices.js)

This happens when typing unreachable JS code in editor (e.g. after `return` statement).

This is accompanied by the following error in console:

```
Uncaught Error: Debug Failure. False expression: token and statement should start at the same point

Error: Debug Failure. False expression: token and statement should start at the same point
    at doChange (ts.worker.bundle.js:135562)
    at ts.worker.bundle.js:135553
    at Function.../../node_modules/monaco-editor/esm/vs/language/typescript/lib/typescriptServices.js.ChangeTracker.with (ts.worker.bundle.js:129852)
    at Object.getCodeActions (ts.worker.bundle.js:135553)
    at ts.worker.bundle.js:130893
    at Object.flatMap (ts.worker.bundle.js:11270)
    at Object.getFixes (ts.worker.bundle.js:130893)
    at ts.worker.bundle.js:141576
    at Object.flatMap (ts.worker.bundle.js:11270)
    at Object.getCodeFixesAtPosition (ts.worker.bundle.js:141574)
    at errors.js:12
```

**monaco-editor version:** 0.20.0
**Browser:** Chrome
**OS:** Windows 10
**Playground code that reproduces the issue:**

We use an ESM version of Monaco editor (imported as a module in our TS code), minified code seems not to have this `debugger` keyword in it.

**Comment by alexdima** [comment]
I can see this exists in the file (both the AMD and the ESM variants) around line 2218:

https://github.com/microsoft/monaco-typescript/blob/778ace10c0b702197eb06a3a1dc3b62b46eb5ceb/src/lib/typescriptServices.js#L2218-L2225

```
        function fail(message, stackCrawlMark) {
            debugger;
            var e = new Error(message ? "Debug Failure. " + message : "Debug Failure.");
            if (Error.captureStackTrace) {
                Error.captureStackTrace(e, stackCrawlMark || fail);
            }
            throw e;
        }
```

This file comes from the TS project and it looks to me that this is some kind of assertion failure that you run into. It must be that `debugger;` statement is left behind by them to make it easier to troubleshoot things.

But I agree that should not appear in a shipped library. So I think there are two issues:

1. something in your code snippet triggers an assertion failure.
2. the file ships with a `debugger;` statement which is not professional.

Unfortunately, this belongs to a different project. So please attempt to reproduce with a code snippet at https://www.typescriptlang.org/play/index.html and create an issue directly against the TypeScript project at https://github.com/microsoft/typescript

**Comment by martinkuchar** [comment]
Hi,

I found two texts which leads to this "Uncaught Error: Debug Failure." issue after entering them into the Monaco Editor Playground.
I tried to reproduce it also on TypeScript Playground, but without success. Maybe the reason is that TS version used in TypeScript Playground is 3.9.2 while Monaco Editor Playground uses TS 3.7.5.

Paste the text into the Playground and then just select the text by pressing CTRL+A and check the console.

First string:
`<IMG SRC=j&#X41vascript:alert('test2')>`

Second text:
```
window.addEventListener(“message”, callback, true);
function callback(e) {
        /* process message (e.data) */
}
```

Problem with the first text can be solved by using
```
monaco.languages.typescript.javascriptDefaults.setCompilerOptions({
    ...
    allowUnusedLabels: true,
    ...
});
```
I don't have solution for the second one yet.
